apps/backend/src/lib/broadcast-listener.ts

```diff
@@ -27,6 +33,36 @@ export function createBroadcastListener(
     handler: async (event) => {
       const { workspaceId } = event.payload
 
+      // Author-scoped events: only emit to the author's sockets
+      //
+      // PERFORMANCE NOTE: This iterates all sockets in the stream room (O(n) where n = connected users).
+      // Acceptable for streams with <100 concurrent users. For larger streams, consider adding user-specific
+      // rooms (e.g., `ws:${workspaceId}:user:${userId}`) and emitting directly to those instead of filtering.
+      // Threshold to consider: ~50-100 concurrent users per stream with frequent author-scoped events.
+      if (isAuthorScopedEvent(event)) {
+        const payload = event.payload as
+          | CommandDispatchedOutboxPayload
+          | CommandCompletedOutboxPayload
+          | CommandFailedOutboxPayload
+        const { streamId, authorId } = payload
+        const room = `ws:${workspaceId}:stream:${streamId}`
+
+        // Find all sockets in the stream room and filter by author
+        const sockets = await io.in(room).fetchSockets()
```

Comment:
I'm curious, would it be possible to keep a lookup in-memory somehow? Since sockets are tied to the running machine, we should be able to handle dis/connects and keep track of users in an in-memory map so that we don't to check all sockets, I think?

So like:
```pseudo
socket.on('connect', (args) => {
  // auth stuff
  userId = args.somehow.we.get.the.userId
  this.userSockets[userId] = socket // Should this be a list?
})

// and somehow clean up on disconnect
```